igor-web/src/main/groovy/com/netflix/spinnaker/igor/scm/github/CommitController.groovy

```diff
@@ -50,14 +52,14 @@ class CommitController extends AbstractCommitController {
 
         try {
             commitsResponse = master.gitHubClient.getCompareCommits(projectKey, repositorySlug, requestParams.to, requestParams.from)
-        } catch (RetrofitError e) {
-            if(e.getKind() == RetrofitError.Kind.NETWORK) {
-                throw new NotFoundException("Could not find the server ${master.baseUrl}")
-            } else if(e.response.status == 404) {
-                return getNotFoundCommitsResponse(projectKey, repositorySlug, requestParams.to, requestParams.from, master.baseUrl)
-            }
-            log.error("Unhandled error response, acting like commit response was not found", e)
+        } catch (SpinnakerServerException e) {
+          if(e instanceof  SpinnakerNetworkException) {
+            throw new NotFoundException("Could not find the server ${master.baseUrl}")
+          } else if(e instanceof SpinnakerHttpException && ((SpinnakerHttpException)e).getResponseCode() == 404) {
             return getNotFoundCommitsResponse(projectKey, repositorySlug, requestParams.to, requestParams.from, master.baseUrl)
+          }
+          log.error("Unhandled error response, acting like commit response was not found", e)
+          return getNotFoundCommitsResponse(projectKey, repositorySlug, requestParams.to, requestParams.from, master.baseUrl)
```

Comment:
Before this PR, getNotFoundCommitsResponse was only called on http 404, right?  Let's take this line out.